device/sycl/src/fitting/fitting_algorithm.sycl

```diff
@@ -82,10 +83,10 @@ track_state_container_types::buffer fitting_algorithm<fitter_t>::operator()(
         .submit([&](::sycl::handler& h) {
             h.parallel_for<kernels::fit>(
                 trackParamsNdRange,
-                [det_view, navigation_buffer, track_candidates_view,
+                [det_view, this, navigation_buffer, track_candidates_view,
```

Comment:
Wait... 🤔 This doesn't actually work, does it? 😕 You're trying to pass the `this` pointer to "device code" here. That's no bueno. You'll need to pass the `m_cfg` object by value to this lambda, like all the other parameters are.

Reply:
Maybe you are right. I don't know if it is working because I currently can not compile SYCL due to the local accessor issue :p

Reply:
hmm both m_cfg and this->m_cfg do not compile in the CI

Reply:
Then there's an even deeper issue here. :frowning:

Keeping the code like this is definitely bad. So we need to then understand why `m_cfg` can not be taken by value. I suspect that it either cannot be properly copied/assigned, or it doesn't have a default constructor. Both of which are needed for proper SYCL operation. :thinking:

Reply:
I will take a look once I can compile with SYCL

Reply:
I confirmed that the code didn't work with `this`:
Running Seeding on device: NVIDIA GeForce RTX 2070
```
PI CUDA ERROR:
        Value:           700
        Name:            CUDA_ERROR_ILLEGAL_ADDRESS
        Description:     an illegal memory access was encountered
        Function:        wait
        Source Location: /home/beomki/packages/llvm/sycl/plugins/cuda/pi_cuda.cpp:449
```


But still passing `m_cfg` does not compile with the following error message:
```
/home/beomki/projects/traccc/traccc/device/sycl/src/fitting/fitting_algorithm.sycl:91:43: error: 'this' cannot be implicitly captured in this context
                                          m_cfg, navigation_buffer,
                                          ^
```                                          

So I copied `m_cfg` into another parameter and passed it to the kernel which worked